### ISSUE TITLE
When no parashat because of a holiday, display the name of the holiday in the parashat field.

### ISSUE THREAD

**Issue by gzh1960** [sgladstone/com.fountaintribe.hebrewcalendarhelper#12]
Shabbat Before yahrzeit not displayed due to Festival of Pesach.

Have been updating our records and found that the 'shabbat before the yahrzeit date' was empty for the yahrzeit date of 20th Nisan 5777 (Sunday 16th April 2017). The 'shabbat after yahrzeit contains the correct value.

I am assuming this is a hebcal issue but it would be great if we could cater for this somehow and put in e.g. Pesach 5 CH"M for Sat 15th April 2017 (19 Nisan 5777). This logic would need to apply for other festivals which fall on the shabbat during the year too.

**Comment by sgladstone** [comment]
In my environments,  I set up a deceased contact with the English date of death "April 4, 2010" (death occurred before sunset.)  In the custom search,  the fields "Friday Night Before Yahrzeit"  and "Saturday Morning Before Yahrzeit" are populated correctly.  (ie  April 14, 2017 April 15, 2017) 

I did notice the field "Parashat for Shabbat Before" is empty in this situation. This makes sense as there is no parashat on that shabbat,  See

http://www.hebcal.com/hebcal/?v=1&geo=geoname&geonameid=5128581&m=50&lg=s&month=x&year=2017&c=on&s=on&maj=on&min=on&mod=on&mf=on&ss=on&nx=on#cal-2017-04

Is this what you are experiencing?

**Comment by gzh1960** [comment]
Sarah, sorry I meant 'Parsha for shabbat before'. Dates are ok just the parsha is empty. Is it possible to have it populated with Pesach 5 CH"M and other vaues for other festivals which fall on shabbat e.g. Succot 2?

**Comment by sgladstone** [comment]
Its possible, but I am going to categorize this as an enhancement request.

**Comment by gzh1960** [comment]
no worries thanks for your help on this one.

**Comment by sgladstone** [comment]
Dealing with holidays may require the addition of a configuration screen for this extension.  For example, some people observe 2 days of yuntiff at the beginning of Pesach, some observe 1 day of yuntiff. Same issue with Rosh Hashana: Some people observe 1 day, others observe 2 days. 

Also people who observe 2 days outside of Israel, will only observe 1 day in Israel.  In Israel the 1 day observance is the standard for all Israelis, no matter how traditional or progressive their specific synagogue is.

**Comment by sgladstone** [comment]
This is done. In the "yahrzeit search" results, holiday is in its own column.    In the mail merge token, if parashat is empty then the holiday name is used instead.

**Comment by gzh1960** [comment]
Have tested and only thing I could find was that the tokens are missing and the value is not in the column that I expected, see attachment.
[Issue12 after changes.pdf](https://github.com/sgladstone/com.fountaintribe.hebrewcalendarhelper/files/610469/Issue12.after.changes.pdf)

**Comment by sgladstone** [comment]
Garry -

In the custom search result columns for "Yahrzeit Search", there are
separate columns for parasha and holiday. I did this because there is
plenty of room here to add extra columns, and sometimes there is a holiday
and a parasha.

If you use the mail merge tokens {yahrzeit.parashat_shabbat_before___day_7}
or  {yahrzeit.parashat_shabbat_after___day_7}    those will display the
holiday if there is no parasha.   I updated the wiki to make this more
clear:
https://github.com/sgladstone/com.fountaintribe.hebrewcalendarhelper/wiki
